File: 프로그래머스/Level 1/가장 가까운 같은 글자.py
# https://school.programmers.co.kr/learn/courses/30/lessons/142086?language=python3
def solution(s):
    answer = []
    check = {}
    
    for i in range(len(s)):
        if not s[i] in check: # 처음 등장
            answer.append(-1)
        else: # 있다면
            answer.append(i-check[s[i]])
        check[s[i]] = i # 교체
    return answer

File: 프로그래머스/Level 1/가장 가까운 같은 글자.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/142086?language=swift
import Foundation

func solution(_ s:String) -> [Int] {
    var answer:[Int] = []
    var s = s.map {String($0)}
    var cmp:[String] = []
    
    for i in s.enumerated() {
        if cmp.contains(i.1) == true {
            let targetIndex = cmp.lastIndex(of: i.1)!
            answer.append(i.0 - targetIndex)
        } else { // 처음 나온 경우
            answer.append(-1)
        }
        cmp.append(i.1)
    }

    return answer
}